### PR TITLE
RHDEVDOCS-5918: Content creation for the ApplicationSet commands

### PR DESCRIPTION
**Pull request by Dhruv-Soni11**
Version(s):

Issue:

GitOps 1.12

https://issues.redhat.com/browse/RHDEVDOCS-5918

Link to docs preview:

https://74591--ocpdocs-pr.netlify.app/openshift-gitops/latest/gitops_cli_argocd/argocd-gitops-cli-reference#argocd-appset-create_argocd-gitops-cli-reference
https://74591--ocpdocs-pr.netlify.app/openshift-gitops/latest/gitops_cli_argocd/argocd-gitops-cli-reference#argocd-appset-delete_argocd-gitops-cli-reference
https://74591--ocpdocs-pr.netlify.app/openshift-gitops/latest/gitops_cli_argocd/argocd-gitops-cli-reference#argocd-appset-get_argocd-gitops-cli-reference
https://74591--ocpdocs-pr.netlify.app/openshift-gitops/latest/gitops_cli_argocd/argocd-gitops-cli-reference#argocd-appset-list_argocd-gitops-cli-reference
https://74591--ocpdocs-pr.netlify.app/openshift-gitops/latest/argocd_application_sets/managing-app-sets-in-non-control-plane-namespaces

SME review: anjoseph@redhat.com @reginapizza 
QE review: @varshab1210 
Internal peer review: @eromanova97 
Peer review: 

QE review:
- [ ] QE has approved this change.

Additional information:

### CODE REVIEW

**Comment by anandf** [comment]
Can we have a separate section for AppSet related commands. We have a section for `Utility Commands` can We have a similar section for AppSet related commands. Right now, I see that its placed under `Global Options` which is not correct. Can we have a section titled `Application Set management commands`. Refer this link. https://github.com/redhat-developer/gitops-operator/blob/b3281fd6e56679bfd723bcef8bc3626b86cf7251/docs/OpenShift%20GitOps%20CLI%20User%20Guide.md

File: modules/ref_argocd-appset-create.adoc [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+[source,terminal]
+----
+argocd appset create [flags]
+----
+
+.Example - Create a new filename or a URL for an `ApplicationSet` resource using the following command:
+
+[source,terminal]
+----
+Create ApplicationSets
```
Comment (anandf) [inline modules/ref_argocd-appset-create.adoc:21]: `Create ApplicationSets` should not be part of the command

File: modules/ref_argocd-appset-delete.adoc [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+.argocd appset create flags
+[options="header",cols="1,1"]
+|===
+|Flag |Description
+
+|`-h, --help`
+|Help output for the `argocd appset delete` subcommand.
+
+|`-y, –yes`
+|An `ApplicationSet` resource can have many applications as child resources. So, if you delete an `ApplicationSet` resource, it means that an application is also deleted. If you use the `-y` flag, then the command does not prompt for confirmation and deletes each child application resource.
```
Comment (anandf) [inline modules/ref_argocd-appset-delete.adoc:34]: `An `ApplicationSet` resource can have one or more applications associated to it as child resources.`
`it means applications associated with the Application set are also deleted`
`and inturn deletes all the applications associated to the ApplciationSet`

File: modules/ref_argocd-appset.adoc [hunk 3]
```diff
@@ -0,0 +1,16 @@
+// Module is included in the following assemblies:
+//
+// * gitops_cli_argocd/argocd-gitops-cli-reference.adoc
+
+:_mod-docs-content-type: REFERENCE
+
+[id="argocd-appset_{context}"]
+= argocd appset
+
+This section lists the basic `argocd appset` commands. The `argocd appset` command allows you to manage `ApplicationSet` resources for your application.
```
Comment (anandf) [inline modules/ref_argocd-appset.adoc:10]: remove the last part `for your application`

File: modules/ref_argocd-appset-create.adoc (hunk 1 above)
Comment (Dhruv-Soni11) [reply modules/ref_argocd-appset-create.adoc:21]: Fixed.

File: modules/ref_argocd-appset-delete.adoc (hunk 2 above)
Comment (Dhruv-Soni11) [reply modules/ref_argocd-appset-delete.adoc:34]: Fixed.

**Review by eromanova97 — COMMENTED** [review]
Hello @Dhruv-Soni11 I finished my initial review and left some comments. Thank you!

File: gitops_cli_argocd/argocd-gitops-cli-reference.adoc [hunk 4]
```diff
@@ -17,6 +17,21 @@ include::modules/gitops-argocd-cli-basic-syntax.adoc[leveloffset=+1]
 // Global options
 include::modules/gitops-argocd-cli-global-options.adoc[leveloffset=+1]
 
+// argocd appset
+include::modules/ref_argocd-appset.adoc[leveloffset=+1]
+
```
[the remark this answers is not in the document]
Comment (Dhruv-Soni11) [reply gitops_cli_argocd/argocd-gitops-cli-reference.adoc:22]: Agree with your feedback @eromanova97, the descriptive title adds better readability and context.

File: modules/ref_argocd-appset-create.adoc [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+----
+argocd appset create [flags]
+----
+
+.Example - Create a new filename or a URL for an `ApplicationSet` resource using the following command:
+
+[source,terminal]
+----
+argocd appset create <filename or URL> (<filename or URL>...)
+----
```
[the remark this answers is not in the document]
Comment (Dhruv-Soni11) [reply modules/ref_argocd-appset-create.adoc:22]: I added the content from the following upstream content [link](https://github.com/redhat-developer/gitops-operator/blob/06994c99c9b1de8c2dd64c5933617130702b356f/docs/cli/argocd_appset_create.md). And I agree that we have kept the replaceable values in the [Utility commands](https://docs.openshift.com/gitops/1.12/gitops_cli_argocd/argocd-gitops-cli-reference.html#gitops-argocd-cli-utility-commands_argocd-gitops-cli-reference) section. I will tag @Srivaralakshmi as I think we would benefit from her additional guidance.

[the remark this answers is not in the document]
Comment (Dhruv-Soni11) [reply modules/ref_argocd-appset-create.adoc:22]: Fixed.

File: modules/ref_argocd-appset-create.adoc [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+[source,terminal]
+----
+argocd appset create [flags]
+----
+
+.Example - Create a new filename or a URL for an `ApplicationSet` resource using the following command:
+
+[source,terminal]
+----
+argocd appset create <filename or URL> (<filename or URL>...)
```
[the remark this answers is not in the document]
Comment (Dhruv-Soni11) [reply modules/ref_argocd-appset-create.adoc:21]: I will change the syntax based on the reply provided in the following [comment](https://github.com/openshift/openshift-docs/pull/74591#discussion_r1566787097).

File: modules/ref_argocd-appset-delete.adoc [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+----
+argocd appset delete [flags]
+----
+
+.Example - Delete an existing `ApplicationSet` resource using the following command:
+
+[source,terminal]
+----
+argocd appset delete APPSETNAME (APPSETNAME...)
+----
```
[the remark this answers is not in the document]
Comment (Dhruv-Soni11) [reply modules/ref_argocd-appset-delete.adoc:23]: Fixed.

File: modules/ref_argocd-appset-delete.adoc [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+[options="header",cols="1,1"]
+|===
+|Flag |Description
+
+|`-h, --help`
+|Help output for the `argocd appset delete` subcommand.
+
+|`-y, –yes`
+|An `ApplicationSet` resource can have one or more applications associated with it as child resources. If you delete an `ApplicationSet` resource, it means that applications associated with the `ApplicationSet` are also deleted and deletes all the applications associated to the ApplciationSet.
+|===
```
[the remark this answers is not in the document]
Comment (Dhruv-Soni11) [reply modules/ref_argocd-appset-delete.adoc:35]: Fixed.

File: modules/ref_argocd-appset-get.adoc [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+
+[id="argocd-appset-get_{context}"]
+= argocd appset get
+
+Get details of any `ApplicationSet` resource.
+
+.Command syntax
+[source,terminal]
+----
+argocd appset get APPSETNAME [flags]
```
[the remark this answers is not in the document]
Comment (Dhruv-Soni11) [reply modules/ref_argocd-appset-get.adoc:15]: Fixed.

File: modules/ref_argocd-appset-get.adoc [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+----
+
+.Example - Search and print the details of an `ApplicationSet` resource using the following command:
+
+[source,terminal]
+----
+argocd appset get APPSETNAME
+----
```
[the remark this answers is not in the document]
Comment (Dhruv-Soni11) [reply modules/ref_argocd-appset-get.adoc:23]: Yes, I agree that it does not add much value. Can I add better examples that are doing some other task mentioned in the flag table like fetching parameters, --show-params?

File: modules/ref_argocd-appset-delete.adoc [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+----
+
+.Example - Delete an existing `ApplicationSet` resource using the following command:
+
+[source,terminal]
+----
+argocd appset delete APPSETNAME (APPSETNAME...)
+----
+
+.argocd appset create flags
```
[the remark this answers is not in the document]
Comment (Dhruv-Soni11) [reply modules/ref_argocd-appset-delete.adoc:25]: Fixed.

File: modules/ref_argocd-appset-get.adoc [hunk 12]
```diff
@@ -0,0 +1,38 @@
...
+
+.argocd appset create flags
```
[the remark this answers is not in the document]
Comment (Dhruv-Soni11) [reply modules/ref_argocd-appset-get.adoc:25]: Fixed.

File: modules/ref_argocd-appset-get.adoc [hunk 13]
```diff
@@ -0,0 +1,38 @@
...
+|Flag |Description
+
+|`-h, --help`
+|Help output for the `argocd appset get` subcommand.
+
+|`--o, --output string`
+|Select the output format for an `ApplicationSet` resource, such as, JSON, YAML, or Wide. The default format is set to Wide.
+
+|`--show-params`
+|Displays the parameters in an `ApplicationSet` resource
```
[the remark this answers is not in the document]
Comment (Dhruv-Soni11) [reply modules/ref_argocd-appset-get.adoc:37]: Fixed.

File: modules/ref_argocd-appset-list.adoc [hunk 14]
```diff
@@ -0,0 +1,44 @@
...
+----
+
+.Example - List `ApplicationSet` resources available in a namespace using the following command:
+
+[source,terminal]
+----
+argocd appset list
+----
+
+.argocd appset create flags
```
[the remark this answers is not in the document]
Comment (Dhruv-Soni11) [reply modules/ref_argocd-appset-list.adoc:25]: Fixed.

File: modules/ref_argocd-appset-list.adoc [hunk 15]
```diff
@@ -0,0 +1,44 @@
...
+
+View the list of `ApplicationSet` resources in a namespace.
+
+.Command syntax
+[source,terminal]
+----
+argocd appset list [flags]
+----
+
+.Example - List `ApplicationSet` resources available in a namespace using the following command:
```
[the remark this answers is not in the document]
Comment (Dhruv-Soni11) [reply modules/ref_argocd-appset-list.adoc:18]: Fixed.

File: modules/ref_argocd-appset-list.adoc [hunk 16]
```diff
@@ -0,0 +1,44 @@
...
+|Flag |Description
+
+|`-h, --help`
+|Help output for the `argocd appset list` subcommand.
+
+|`--o, --output string`
+|Select the output format for an `ApplicationSet` resource, such as, JSON, YAML, or Wide. The default format is set to Wide.
+
+|`-N, --appset-namespace string`
+|List only `ApplicationSet` in a namespace
```
[the remark this answers is not in the document]
Comment (Dhruv-Soni11) [reply modules/ref_argocd-appset-list.adoc:37]: Fixed.

File: modules/ref_argocd-appset-list.adoc [hunk 17]
```diff
@@ -0,0 +1,44 @@
...
+
+|`-p, --project stringArray`
+|Filter `ApplicationSet` by the name of a project
```
[the remark this answers is not in the document]
Comment (Dhruv-Soni11) [reply modules/ref_argocd-appset-list.adoc:40]: Fixed.

File: modules/ref_argocd-appset-list.adoc [hunk 18]
```diff
@@ -0,0 +1,44 @@
...
+
+|`-l, --selector string`
+|List `ApplicationSet` by labels
```
[the remark this answers is not in the document]
Comment (Dhruv-Soni11) [reply modules/ref_argocd-appset-list.adoc:43]: Fixed.

File: modules/ref_argocd-appset.adoc [hunk 19]
```diff
@@ -0,0 +1,16 @@
+// Module is included in the following assemblies:
+//
+// * gitops_cli_argocd/argocd-gitops-cli-reference.adoc
+
+:_mod-docs-content-type: REFERENCE
+
+[id="argocd-appset_{context}"]
+= argocd appset
```
[the remark this answers is not in the document]
Comment (Dhruv-Soni11) [reply modules/ref_argocd-appset.adoc:8]: Fixed.

File: argocd_application_sets/managing-app-sets-in-non-control-plane-namespaces.adoc [hunk 20]
```diff
@@ -38,5 +38,9 @@ include::modules/gitops-allowing-scm-providers.adoc[leveloffset=+1]
 * link:https://argo-cd.readthedocs.io/en/release-2.10/operator-manual/applicationset/#the-applicationset-resource[The `ApplicationSet` resource]
 * link:https://argo-cd.readthedocs.io/en/release-2.10/operator-manual/applicationset/Appset-Any-Namespace/[ApplicationSet in any namespace]
 * xref:../argocd_instance/argo-cd-cr-component-properties.adoc#argo-cd-cr-component-properties[Argo CD custom resource and component properties]
+* xref:../gitops_cli_argocd/argocd-gitops-cli-reference.adoc#ref_argocd-appset-create[argocd appset create]
+* xref:../gitops_cli_argocd/argocd-gitops-cli-reference.adoc#ref_argocd-appset-delete[argocd appset delete]
+* xref:../gitops_cli_argocd/argocd-gitops-cli-reference.adoc#ref_argocd-appset-get[argocd appset get]
+* xref:../gitops_cli_argocd/argocd-gitops-cli-reference.adoc#ref_argocd-appset-list[argocd appset list]
```
[the remark this answers is not in the document]
Comment (Dhruv-Soni11) [reply argocd_application_sets/managing-app-sets-in-non-control-plane-namespaces.adoc:44]: Agreed. Fixed.

File: modules/ref_argocd-appset-create.adoc (hunk 5 above)
[the remark this answers is not in the document]
Comment (eromanova97) [reply modules/ref_argocd-appset-create.adoc]: Yes, definitely, thank you! I am curious about the answer as well. :+1:

File: modules/ref_argocd-appset-get.adoc (hunk 10 above)
[the remark this answers is not in the document]
Comment (eromanova97) [reply modules/ref_argocd-appset-get.adoc]: Yes, that is a great idea, I think that would add a lot more value than the current example :+1:

[the remark this answers is not in the document]
Comment (Dhruv-Soni11) [reply modules/ref_argocd-appset-get.adoc]: Added an example. Please confirm if this is fine, @eromanova97.

File: modules/ref_argocd-appset-list.adoc [hunk 21]
```diff
@@ -0,0 +1,44 @@
...
+----
+argocd appset list [flags]
+----
+
+.Example - List `ApplicationSet` resources available in a namespace using the following command:
+
+[source,terminal]
+----
+argocd appset list
+----
```
[the remark this answers is not in the document]
Comment (Dhruv-Soni11) [reply modules/ref_argocd-appset-list.adoc]: Added an example. Please confirm if this is fine, @eromanova97.

[the remark this answers is not in the document]
Comment (eromanova97) [reply modules/ref_argocd-appset-list.adoc]: Yes, this definitely looks better :+1:  thanks!

**Comment by anandf** [comment]
LGTM

**Comment by Dhruv-Soni11** [comment]
Hi @eromanova97,

As per the discussion on this Slack channel [thread](https://redhat-internal.slack.com/archives/C01RQH8KQ87/p1713219396601659) and the Cabal call agenda [document](https://docs.google.com/document/d/1auOZNdeiwGcrms1LupU8aeWkUcGfFJqNFaUyLutMmAQ/edit), the team has decided that the content creation for this activity will be stopped as it does not add many benefits to the existing content in the upstream documentation. 

Due to the constant maintenance that might be required to keep the content accurate in the downstream documentation, it is decided that the best approach for this activity is to refer to the upstream documentation for now.

I am closing this thread for now due to these concerns.

**Pull request closed without merging**